### PR TITLE
Kiting / delayed evade fixup

### PR DESCRIPTION
**Pull request by nemesus**
Modified mob behavior to use the radial aggro of the last point of attack (direct and initial DOT tick).

Reason:  Ran into a couple instances where the mob that I was attacking would prematurely disengage even when the character was within combat range.

Example: ran into the (cave-)yeti cave in hillsbrad foothills, attacked and killed 4-5 yet one yeti from the inside the cave paced, took one DOT and evaded.

### CODE REVIEW

**Comment by DDuarte** [comment]
Thanks for the PR :+1: 

Before we discuss the actual code changes, can you fix the code style? http://collab.kpsn.org/pages/viewpage.action?pageId=10682946 namely braces, indentation and spaces between parenthesis.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 1]
```diff
@@ -751,7 +751,7 @@ void AuraEffect::Update(uint32 diff, Unit* caster)
             m_periodicTimer -= diff;
         else // tick also at m_periodicTimer == 0 to prevent lost last tick in case max m_duration == (max m_periodicTimer)*N
         {
-            ++m_tickNumber;
+            if (++m_tickNumber == 0) m_tickNumber = 2; // Only first go can be 1
```
Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: this doesnt look like its a part of this

Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: wtf?

Comment (nemesus) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: I seriously doubt a single DOT would survive longer than 4,294,967,295 ticks, however this check marries with the ones below to ensure that only the first tick is counted

**Comment by Shauren** [comment]
So you reset the "too far from home" position for distance check on damage event but this is not blizzlike - old code was correct here, evade if not damaged for 10s (except dot ticks) when too far from spawn point

**Comment by nemesus** [comment]
That was not the behavior I remember, direct and initial DOT application affected the mobs 'interest' in you.  What you're suggesting as proper, sounds detrimental to a DOT spec'd class; they would be unable to hold a tag on multiple mobs outside their 'home range' unless they steamrolled them (such as my example the effected mob disengaged despite having a tasty target right in front of it).

**Comment by Shauren** [comment]
of course initial dot application resets the timer

**Pull request closed without merging**